client/src/app/shared/components/os-sort-filter-bar/os-sort-filter-bar.component.ts

```diff
@@ -0,0 +1,388 @@
+ * filters.
+ *
+ * ## Examples:
+ * ### Usage of the selector:
+ *
+ * ```html
+ * <os-sort-filter-bar model="agenda" [modelRepo]="repo.getViewModelListObservable()"
+ *      [filters]="filterOptions" [sorting]="sortOptions"
+ *      (filteredData)="onFilterData($event)">
+ * </os-sort-filter-bar>
+ * ```
+ * Then, write some filterOptions and sortOptions into the listview.
+ */
+@Component({
+    selector: 'os-sort-filter-bar',
+    templateUrl: './os-sort-filter-bar.component.html',
+    styleUrls: ['./os-sort-filter-bar.component.scss']
+})
+export class OsSortFilterBarComponent<V extends BaseViewModel> implements OnInit {
+
+    @Input()
+    public model: string;
+
+    /**
+     * The definition of the sorting columns to be made available
+     */
+    @Input()
+    public sorting: OsSortingDefinition<V>;
+
+    /**
+     * A list of filters to be made available
+     */
+    @Input()
+    public filters: OsFilter[];
+
+    @ViewChild(MatMenuTrigger)
+    public menuTrigger: MatMenuTrigger;
+
+    /**
+     * The original repository where the data are fetched from
+     */
+    @Input()
+    public modelRepo: Observable<V[]>;
+
+    /**
+     * The filter side drawer
+     */
+    @ViewChild('filterMenu')
+    public filterMenu: FilterMenuComponent;
+
+    /**
+     * The bottom sheet used to alter sorting in mobile view
+     */
+    @ViewChild('sortBottomSheet')
+    public sortBottomSheet: OsSortBottomSheetComponent<V>;
+
+    /**
+     * An event emitter to offer filtered Data to display.
+     */
+    @Output()
+    public filteredData = new EventEmitter<MatTableDataSource<V>>();
+
+    /**
+     * The 'opened/active' state of the fulltext filter input field
+     */
+    public isSearchBar = false;
+
+    /**
+     * Helper data source to assist with MatSort sorting and fulltext filter
+     * @ignore
+     */
+    private _filteredDataSource = new MatTableDataSource<V>();
+
+    /**
+     * The current data that come from the repository
+     */
+    private _originalData: V[];
+
+    public constructor(
+        private bottomSheet: MatBottomSheet,
+        public translate: TranslateService,
+        public vp: ViewportService,
+        private storageService: StorageService
+        ) {
+            this.filterMenu = new FilterMenuComponent(translate);
+            this.filterMenu.activeFilters.subscribe(newFilters => {
+                this.filters = newFilters;
+                this.filterData();
+            });
+    }
+
+    public ngOnInit(): void {
+        this.checkStoredFilterSort();
+        if (this.modelRepo){
+            this.modelRepo.subscribe(originalData => {
+                this._originalData = originalData;
+                this.filterData();
+            });
+        }
+    }
+
+    /**
+     * Iterates through the filters, sorts and emits the new filtered data through {@link filteredData}
+     */
+    private filterData(): void {
```

Comment:
There are several ways to address this.
One might store the sort and filter information in the models, the view models, here or in a dedicated service.
I would advice to put that inside of a core service to handle the sorting and filtering.